server.go

```diff
@@ -893,6 +942,68 @@ func (s *Server) applyCreateShardGroupIfNotExists(m *messaging.Message) (err err
 	}
 
 	return
+
+}
+
+// CreateShardGroupIfNotExists creates the shard group for a retention policy for the interval a timestamp falls into.
+func (s *Server) CreateShardGroupIfNotExists(database, policy string, timestamp time.Time) error {
+	c := &createShardGroupIfNotExistsCommand{Database: database, Policy: policy, Timestamp: timestamp}
+	_, err := s.broadcast(createShardGroupIfNotExistsMessageType, c)
+	return err
+}
+
+func (s *Server) applyCreateShardGroupIfNotExists(m *messaging.Message) (err error) {
+	var c createShardGroupIfNotExistsCommand
+	mustUnmarshalJSON(m.Data, &c)
+
+	// Retrieve database.
+	db := s.databases[c.Database]
+	if s.databases[c.Database] == nil {
+		return ErrDatabaseNotFound
+	}
+
+	// Validate retention policy.
+	rp := db.policies[c.Policy]
+	if rp == nil {
+		return ErrRetentionPolicyNotFound
+	}
+
+	return s.createShardGroupIfNotExists(m.Index, db, rp, c.Timestamp)
+}
+
+// PreCreateShardGroupIfNotExists creates the shard group for a retention policy for the interval a timestamp falls into.
```

Comment:
This comment does not appear to be correct. There is no reference to "timestamp" in this function.